SPIE2016-9913-16/9913-16.tex

```diff
@@ -0,0 +1,303 @@
+To get this working on PyPI it is necessary either to generate the \texttt{setup.py} using a SCons target or else use EUPS directly in the \texttt{setup.py}.
+The latter would require that EUPS is also available from PyPI (it currently is not) but might be less prone to errors.
+
+\subsection{Astronomical Coordinates}
+
+The LSST \texttt{afw.coord} package provides coordinate conversions between ICRS, FK5, Galactic, ecliptic and topocentric coordinate systems.
+It is written in C++ and has no external dependencies on standard coordinate libraries.
+The \texttt{astropy.coordinates} package provides a very usable Python API on top of the standard ERFA astrometry C library\footnote{ERFA is a relicensed version of the standard SOFA library\cite{2011SchpJ...611404H}.}.
+It supports all the coordinate systems provided by the LSST package and many more including FK4.
+
+The LSST C++ code does not use any system other than ICRS therefore we are proposing to remove this package and replace it with a simple spherical coordinates class.
+All the Python code will then be modified to use \texttt{astropy.coordinates} and at the Python/C++ boundary coordinates will be converted to/from ICRS.
+
+\subsection{Tables}
+
+LSST and Astropy both have table libraries, but these have different data models and hence different strengths.
+LSST's are better for row-based access and provide a limited ORM layer that's valuable for pipeline use when the columns are known in advance and data are appended to the table, while Astropy's excel at column-based access and use in interactive analysis.
+Furthermore, Astropy tables now support export to pandas DataFrames\cite{mckinney-proc-scipy-2010}, opening up a richer environment for query-based analysis.
+We anticipate being able to do a lot by just creating Astropy views to LSST table objects, thereby not doubling the memory requirements, and we have been able to put together a proof-of-concept implementation for doing this.
+This will make LSST tables far more accessible to general users and provide a far richer ecosystem of visualization and debugging tools than would be possible if LSST were required to write their own code.
+
+\subsection{C++ Bindings}
+
+The LSST Data Management team adopted SWIG as the standard method for providing a Python interface to the C++ code.
+Astropy has adopted Cython\cite{2010/content/aip/journal/cise/13/2/10.1109/MCSE.2010.118} when it is required to call C code.
+The Astropy project has no hard rules concerning how C/C++ code should be included in affiliate package but has a strong preference for Cython.
+LSST is not averse to switching away from SWIG (which results in massive code bloat and extended compile times as currently implemented) and has agreed to investigate Cython\cite{dmtn-013} and pybind11\cite{dmtn-014}\footnote{\url{http://pybind11.readthedocs.org/}}.
```

Comment:
I'll leave it in though as I cite both the DMTNs and we haven't explicitly written cython off yet.